pkg/workflow/argo.go

```diff
@@ -561,7 +561,7 @@ func (a *argo) generateSubchartAndAppDAGTasks(ctx context.Context, g *v1alpha1.A
 		Spec: fluxhelmv2beta1.HelmReleaseSpec{
 			Chart: fluxhelmv2beta1.HelmChartTemplate{
 				Spec: fluxhelmv2beta1.HelmChartTemplateSpec{
-					Chart:   app.Spec.Chart.Name,
+					Chart:   pkg.ConvertToDNS1123(app.Spec.Chart.Name),
```

Comment:
What happens if the chart name has special characters like `$, %, etc.`?

Reply:
Maybe I can leverage something like this - https://gitlab.com/gitlab-org/gitlab-runner/-/blob/0e2ae0001684f681ff901baa85e0d63ec7838568/executors/kubernetes/util.go#L268

Would need to make a few adjustments like converting `_` to `-` instead of just replacing it with `""` to trim those chars?